app/controllers/search_controller.rb

```diff
@@ -4,8 +4,14 @@ class SearchController < ApplicationController
   def index
     mail = params[:search][:email]
 
-    @matching_users = User.where.not(id: current_user.id).all
-    @matching_users = @matching_users.where(email: mail.strip) if mail.present?
+    if mail.present? && mail != current_user.email
```

Comment:
So, if the user searches for themselves you want it to allies rather than show no results?